### PR TITLE
[BugFix] RLE encoder stuck in FlushRepeatedRun when repeat_count_ >= 0x40000000

### LINKED ISSUE #6513: RLE encoder stuck when repeat_count_ >= 0x40000000
### Steps to reproduce the behavior (Required)
1.  CREATE a RLE encoder 
2.  Put into 0x40000000 same value
3.  Flush

### Expected behavior (Required)
See flush successful

### Real behavior (Required)
Flush stuck in dead loop

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
a328b681ec277ae502d04d168039e59d8538a953

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6513

## Problem Summary(Required) ：

when repeat_count_ >= 0x40000000, repeat_count_ << 1 | 0 is a negative number(indicator_value).
In PutVlqInt function, indicator_value >>= 7 is a logical right move, it will move the right 7 value out
and append 1 at the beginning of the number, finally lead to a dead loop.

### CODE REVIEW

File: be/src/util/bit_stream_utils.inline.h [hunk 1]
```diff
@@ -86,11 +86,14 @@ inline void BitWriter::PutAligned(T val, int num_bytes) {
 }
 
 inline void BitWriter::PutVlqInt(int32_t v) {
```
Comment (chaoyli) [inline be/src/util/bit_stream_utils.inline.h:88]: you can use uint32_t as a parameter

File: be/src/util/bit_stream_utils.inline.h [hunk 2]
```diff
@@ -85,10 +85,12 @@ inline void BitWriter::PutAligned(T val, int num_bytes) {
     memcpy(ptr, &val, num_bytes);
 }
 
-inline void BitWriter::PutVlqInt(int32_t v) {
+inline void BitWriter::PutVlqInt(uint32_t v) {
+    int num_bytes = 0;
```
Comment (imay) [inline be/src/util/bit_stream_utils.inline.h:89]: This will cause a unused variable warning when no-debug

**Review by chaoyli — APPROVED** [review]

**Review by imay — APPROVED** [review]
LGTM

File: be/test/util/rle_encoding_test.cpp [hunk 3]
```diff
@@ -158,6 +158,18 @@ TEST(Rle, SpecificSequences) {
     }
 }
 
+TEST(Rle, ISSUE_6513) {
+    std::vector<uint64_t> values;
+
+    values.resize(0x40000000);
```
Comment (trueeyu) [inline be/test/util/rle_encoding_test.cpp:164]: run slowly in daily asan

File: be/src/util/bit_stream_utils.h [hunk 4]
```diff
@@ -77,6 +77,9 @@ class BitWriter {
     // to the next byte boundary.
     void Flush(bool align = false);
 
+    // Maximum byte length of a vlq encoded int
+    static const int MAX_VLQ_BYTE_LEN = 5;
```
Comment (trueeyu) [inline be/src/util/bit_stream_utils.h:81]: why add one new `MAX_VLQ_BYTE_LEN`

Comment (xiaoyong-z) [reply be/src/util/bit_stream_utils.h:81]: In the original code, it is local static variables within another class, i can not use it in current class.

**Pull request merged**